lib/index.js

```diff
@@ -74,21 +76,43 @@ class EvervaultClient {
     const isIgnoreRequest = (domain) =>
       this._isIgnoreRequest(domain, ignoreExact, ignoreEndsWith);
 
-    const initializationPromise = this.http.getCert().then((pem) => {
+    let x509 = null;
+
+    async function update_certificate() {
+      const pem = await this.http.getCert();
+      let cert = pem.toString();
+      x509 = new X509Certificate(cert);
       tls.createSecureContext = (options) => {
         const context = origCreateSecureContext(options);
         context.context.addCACert(pem);
         return context;
       };
-    });
+    }
+
+    function is_certificate_unavaliable() {
+      return x509 == null;
+    }
+
+    function is_certificate_expired() {
+      if (x509 == null) {
+        return false;
+      }
+      let ts = Date.now();
+      let date = new Date(ts);
+      return (
+        date.getUTCDate() > x509.validTo || date.getUTCDate() < x509.validFrom
+      );
+    }
 
     function wrapMethodRequest(...args) {
       const ignore = isIgnoreRequest(args);
       args = args.map((arg) => {
         if (!ignore && arg instanceof Object) {
           arg.agent = new HttpsProxyAgent(
             tunnelHostname,
-            initializationPromise
+            update_certificate,
+            is_certificate_expired,
+            is_certificate_unavaliable
```

Comment:
A couple points on `is_certificate_unavailable` and `is_certificate_expired`
- should be camel case
- `is_certificate_expired` is being passed to the ProxyAgent which is taking it as `isCertificateValidCallback` which is the opposite
- I don't think the logic for `is_certificate_expired` makes sense - if it's undefined, it's false which makes it seem as though an undefined cert is valid, so this function doesn't make sense without `is_certificate_unavailable`

Would be nicer to have `isCertificateValid` which returns false on undefined, and true if the current time is within the validity period of the cert